### PR TITLE
Add animations to blog website

### PR DESCRIPTION
**Pull request by sahadat-sk**
Fixing: [#1664](https://github.com/keploy/keploy/issues/1664)

**Features added**

- Added loading screen animation
- Hover button animation on the 'technology' and 'community' buttons in the landing page
- Hover animation on blog preview

https://github.com/keploy/keploy/assets/98450585/759527c3-b293-4074-8942-6aa781887403

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -1,8 +1,38 @@
...
+import "../styles/index.css";
+import Router from "next/router";
+import { PageLoader } from "../components/PageLoader";
+
+import { AnimatePresence } from "framer-motion";
+import { useEffect, useState } from "react";
 
 function MyApp({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+  const [loading, setLoading] = useState(false);
```
Comment (Hermione2408) [inline pages/_app.tsx:10]: loading by default should be true as it will flicker otherwise

File: pages/_app.tsx [hunk 2]
```diff
@@ -1,8 +1,38 @@
...
+  useEffect(() => {
+    // Used for page transition
+    const start = () => {
```
Comment (Hermione2408) [inline pages/_app.tsx:13]: can rename it to startLoader and instead of end stopLoader

**Review by Hermione2408 — CHANGES_REQUESTED** [review]
Hey @sahadat-sk please reduce loader size

File: pages/_app.tsx (hunk 2 above)
Comment (Hermione2408) [reply pages/_app.tsx:13]: Hey @sahadat-sk, I really appreciate the effort you've put into the hover button – it's a cool feature! I think the preview card itself is quite intuitive and users will naturally know to click on it for more info. Maybe we can do without the extra button to keep things sleek. What do you think?

File: pages/_app.tsx (hunk 1 above)
Comment (sahadat-sk) [reply pages/_app.tsx:10]: Indeed, the loading animation triggers upon route changes. If we were to set it to true initially, the loading screen would consistently display, even when the route isn't changing.

**Comment by sahadat-sk** [comment]
Hey @Hermione2408, can you please review it now?

**Comment by sahadat-sk** [comment]
> > Hey @Hermione2408, can you please review it now?
> 
> Hey @sahadat-sk due to latest PR merge your branch has conflicts, please resolve them

Done!

**Comment by sahadat-sk** [comment]
Hello @Hermione2408, sorry for the repeated linting changes 😅. I've reverted them and will be more careful in the future. Also, I've updated the pull request description with demo videos.

**Review by Hermione2408 — COMMENTED** [review]
@nehagup looks good to me!

**Comment by Hermione2408** [comment]
Hey @sahadat-sk Build is failing for node version 21 
> build
> next build

 ✓ Linting and checking validity of types    
   ▲ Next.js 14.0.1
   - Environments: .env.production

 ✓ Creating an optimized production build    
 ✓ Compiled successfully
   Collecting page data  ..ReferenceError: document is not defined
    at createTag (/home/hermione/Desktop/work/blog-website/node_modules/@lottiefiles/react-lottie-player/dist/lottie-react.js:15:7827)
    at /home/hermione/Desktop/work/blog-website/node_modules/@lottiefiles/react-lottie-player/dist/lottie-react.js:15:20698
    at /home/hermione/Desktop/work/blog-website/node_modules/@lottiefiles/react-lottie-player/dist/lottie-react.js:15:20821
    at /home/hermione/Desktop/work/blog-website/node_modules/@lottiefiles/react-lottie-player/dist/lottie-react.js:15:23614
    at svgNS (/home/hermione/Desktop/work/blog-website/node_modules/@lottiefiles/react-lottie-player/dist/lottie-react.js:15:7504)
    at /home/hermione/Desktop/work/blog-website/node_modules/@lottiefiles/react-lottie-player/dist/lottie-react.js:15:7508
    at createCommonjsModule (/home/hermione/Desktop/work/blog-website/node_modules/@lottiefiles/react-lottie-player/dist/lottie-react.js:15:7169)
    at Object.<anonymous> (/home/hermione/Desktop/work/blog-website/node_modules/@lottiefiles/react-lottie-player/dist/lottie-react.js:15:7397)
    at Module._compile (node:internal/modules/cjs/loader:1368:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1426:10)

> Build error occurred
Error: Failed to collect page data for /community
    at /home/hermione/Desktop/work/blog-website/node_modules/next/dist/build/utils.js:1211:15 {
  type: 'Error'
}

**Comment by sahadat-sk** [comment]
Hey @nehagup, can you check now?

**Review by nehagup — COMMENTED** [review]
@sahadat-sk this looks good! Please resolve the conflicts and we should be ready to merge.

**Review by Hermione2408 — APPROVED** [review]
Hey @nehagup build is working fine LGTM!

**Pull request merged**